nltk/translate/bleu_score.py

```diff
@@ -169,11 +169,17 @@ def corpus_bleu(
         "The number of hypotheses and their reference(s) should be the " "same "
     )
 
+    if weights and isinstance(weights[0], float):
+        weights = [weights]
+    elif isinstance(weights, tuple):
+        weights = [weights]
```

Comment:
I realise I'm being very nitpicky, but this will still not work with e.g. `weights = [0, 1]`. 
My goal is for this to work with any iterable with numbers in it, just like now. With other words, I don't want anyone's current code to break as a result of this feature.

Perhaps a solution is to check the "depth" of weights, i.e. check if we have a nested iterable. 

```suggestion
    # Ensure that weights is a nested list
    try:
        weights[0][0]
    except TypeError:
        weights = [weights]
```

Reply:
@tomaarsen made the changes.. can you see if its fixed in new commit

Reply:
I believe it's fixed! Now `[0, 1]`, `[0.3, 0.7]`, `(0, 1)`, `(0.3, 0.7)` can all still be used like before, alongside the new functionality you've made. 
I've resolved a merge conflict between this PR and changes from #2839. I believe everything works well now!